R/map_nhdplus.R

```diff
@@ -0,0 +1,193 @@
+#' Check for a package
+#' @param pkg package name
+
+check_pkg <- function(pkg) {
+  if (!requireNamespace(pkg, quietly = TRUE))
+    stop("Package '",
+         pkg,
+         "' is required for this functionality, but is not installed. \nTry `install.packages('",
+         pkg,
+         "')`", call. = FALSE)
+}
+
+#' @title Make Interactive Map of NHDPlus
+#' @description Given a list of outlets, get their basin boundaries and network and return a leaflet map in
+#' EPSG:4326.
+#' @inheritParams plot_nhdplus
+#' @return data.frame plot data is returned invisibly in NAD83 Lat/Lon.
+#' @details map_nhdplus supports several input specifications. An unexported function "as_outlet"
+#' is used to convert the outlet formats as described below.
+#' \enumerate{
+#'   \item if outlets is omitted, the bbox input is required and all nhdplus data
+#'   in the bounding box is plotted.
+#'   \item If outlets is a list of integers, it is assumed to be NHDPlus IDs (comids)
+#'   and all upstream tributaries are plotted.
+#'   \item if outlets is an integer vector, it is assumed to be all NHDPlus IDs (comids)
+#'   that should be plotted. Allows custom filtering.
+#'   \item If outlets is a character vector, it is assumed to be NWIS site ids.
+#'   \item if outlets is a list containing only characters, it is assumed to be a list
+#'   of nldi features and all upstream tributaries are plotted.
+#'   \item if outlets is a data.frame with point geometry, a point in polygon match
+#'   is performed and upstream with tributaries from the identified catchments is plotted.
+#' }
+#' @export
+#' @examples
+#' \donttest{
+#' map_nhdplus("05428500")
+#'
+#' map_nhdplus("05428500", streamorder = 2)
+#'
+#' map_nhdplus(list(13293970, 13293750))
+#'
+#' source(system.file("extdata/sample_data.R", package = "nhdplusTools"))
+#'
+#' map_nhdplus(list(13293970, 13293750), streamorder = 3, nhdplus_data = sample_data)
+#' }
+#' @importFrom sf st_transform
+
+map_nhdplus <- function(outlets = NULL, bbox = NULL, streamorder = NULL,
+                        nhdplus_data = NULL, gpkg = NULL,
+                        flowline_only = NULL, plot_config = NULL, overwrite = TRUE, cache_data = NULL) {
+
+  check_pkg("leaflet")
+
+  # I did try gt() and warnings are thrown
+  lt <- function(x){ st_transform(x, '+proj=longlat +datum=WGS84') }
```

Comment:
I like having this in line like this. We should move gt in line as well. gt() also extracts the geometry -- which probably doesn't work with leaflet.